search/query-features-with-arcade-expression/src/main/java/com/esri/samples/query_features_with_arcade_expression/QueryFeaturesWithArcadeExpressionSample.java

```diff
@@ -0,0 +1,198 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.esri.samples.query_features_with_arcade_expression;
+
+import java.util.HashMap;
+
+import javafx.application.Application;
+import javafx.geometry.Point2D;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert;
+import javafx.scene.input.MouseButton;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+
+import com.esri.arcgisruntime.ArcGISRuntimeEnvironment;
+import com.esri.arcgisruntime.arcade.ArcadeEvaluator;
+import com.esri.arcgisruntime.arcade.ArcadeExpression;
+import com.esri.arcgisruntime.arcade.ArcadeProfile;
+import com.esri.arcgisruntime.data.ArcGISFeature;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.layers.Layer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.view.Callout;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.portal.Portal;
+import com.esri.arcgisruntime.portal.PortalItem;
+
+public class QueryFeaturesWithArcadeExpressionSample extends Application {
+
+  private MapView mapView;
+  private ArcGISMap map;
+  private Layer policeBeatsLayer;
+  private final String rpdBeatsLayerName = "RPD Beats  - City_Beats_Border_1128-4500";
+  private final String calloutText = "Crimes in the last 60 days: ";
+
+  @Override
+  public void start(Stage stage) {
+    try {
+      // create stack pane and application scene
+      var stackPane = new StackPane();
+      var scene = new Scene(stackPane);
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Query Features With Arcade Expression Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a portal item and use it to create a map
+      var portal = new Portal("https://www.arcgis.com/");
+      var portalItem = new PortalItem(portal, "539d93de54c7422f88f69bfac2aebf7d");
+      map = new ArcGISMap(portalItem);
+
+      // create a map view and set the map to it
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // set callout properties that never change
+      mapView.getCallout().setLeaderPosition(Callout.LeaderPosition.BOTTOM);
+      mapView.getCallout().setTitle("RPD Beats");
+
+      // wait for the map to finish loading data from portal before accessing its layers
+      map.addDoneLoadingListener(() -> {
```

Comment:
after going through this, there is a neater way to handle this that saves a bit of repeated code here too, see below suggestion.

I think the addition of a `ProgressIndicator` would be good here, as I've noticed the web map can take a while to load sometimes. You can instantiate the progress indicator after the api key set up (you shouldn't need to set it's visibility to true as that is its default value). Then once the map has loaded and you have the policeBeatsLayer, set it to invisible. Then add it to the stackpane before the mapView. 

```java
      // wait for the map to finish loading data from portal before accessing its layers
      map.addDoneLoadingListener(() -> {
        if (map.getLoadStatus() == LoadStatus.LOADED) {
          
          for (Layer operationalLayer : map.getOperationalLayers()) {
            // if the operational layer is the RPD Beats layer, show it on the map and hide the progress indicator
            if (operationalLayer.getName().equals(rpdBeatsLayerName)) {
              policeBeatsLayer = operationalLayer;
              operationalLayer.setVisible(true);
              progressIndicator.setVisible(false);
            } else {
              // if the operational layer is any layer other than the RPD Beats layer, don't show it.
              operationalLayer.setVisible(false);
            }
          }

```

Reply:
Have added this if/else block although the progress indicator would disappear without it - presumably the map was being drawn over it once it loaded?